diff --git a/setup.py b/setup.py
index d1f1de7..1d64d11 100644
--- a/setup.py
+++ b/setup.py
@@ -303,12 +303,11 @@ setup(  # Finally, pass this all along to setuptools to do the heavy lifting.
     setup_requires=[
         "certifi>=2020.06.20",
         "numpy>=1.19",
-        "setuptools_scm>=7",
+        "setuptools_scm>=4",
     ],
     install_requires=[
-        "contourpy>=1.0.1",
         "cycler>=0.10",
-        "fonttools>=4.22.0",
+        "fonttools>=3.29.0",
         "kiwisolver>=1.0.1",
         "numpy>=1.19",
         "packaging>=20.0",
@@ -317,7 +316,7 @@ setup(  # Finally, pass this all along to setuptools to do the heavy lifting.
         "python-dateutil>=2.7",
     ] + (
         # Installing from a git checkout that is not producing a wheel.
-        ["setuptools_scm>=7"] if (
+        ["setuptools_scm>=4"] if (
             Path(__file__).with_name(".git").exists() and
             os.environ.get("CIBUILDWHEEL", "0") != "1"
         ) else []
